docs/src/componentDocs/ChannelValue/playground/PlaygroundPage.tsx

```diff
@@ -1,17 +1,158 @@
+        type: 'number',
+        typeLabel: `number | string`,
+        initialValue: 30,
+        description: 'The size of the font',
+        required: false,
+        minValue: 10,
+        maxValue: 50,
+        valueStep: 10,
+        category: 'Optional Props',
+        defaultValue: 'inherit',
+    },
+    {
+        id: 'color',
+        label: 'color',
+        type: 'color',
+        typeLabel: 'string',
+        description: 'The color of the font',
+        required: false,
+        category: 'Optional Props',
+        defaultValue: 'inherit',
+        allowMuiColors: true,
+    },
+    {
+        id: 'prefix',
+        type: 'boolean',
+        initialValue: false,
+        typeLabel: 'boolean',
+        description: 'Show units before the value',
+        required: false,
+        category: 'Optional Props',
+        defaultValue: false,
+    },
+    {
+        id: 'unitSpace',
+        type: 'select',
+        typeLabel: `'auto' | 'show' | 'hide'`,
+        initialValue: 'auto',
+        options: [
+            { label: 'auto', value: 'auto' },
+            { label: 'hide', value: 'hide' },
+            { label: 'show', value: 'show' },
+        ],
+        description: 'Show/Hide spacing between the value and units',
+        required: false,
+        category: 'Optional Props',
+        defaultValue: 'auto',
+    },
+    // Other Configuration
+    {
+        id: 'htmlColor',
+        type: 'color',
+        initialValue: '#f33333',
+        // typeLabel: 'string',
+        description: 'The color applied to the primary icon',
+        required: false,
+        label: 'Icon Color',
+        category: 'Other Configuration',
+    },
+];
+
+const ChannelValuePreview: PreviewComponent = ({ data }) => {
+    const { htmlColor, icon, ...rest } = data as unknown as ChannelValueProps & { htmlColor: string };
+    const getIcon = (value: string): JSX.Element | undefined => {
+        switch (value) {
+            case '<TrendingUp />':
+                return <TrendingUp htmlColor={htmlColor || 'inherit'} />;
+            case '<TrendingDown />':
+                return <TrendingDown htmlColor={htmlColor || 'inherit'} />;
+            case 'undefined':
+            default:
+                return undefined;
+        }
+    };
+
+    return (
+        <Stack alignItems={'center'} justifyContent={'center'} sx={{ width: '100%', height: '100%' }}>
+            <ChannelValue {...rest} icon={getIcon(icon as unknown as string)} />
+        </Stack>
+    );
+};
+
+const generateSnippet: CodeSnippetFunction = (data) =>
+    `<ChannelValue 
+    ${getPropsToString(getPropsMapping(data, inputConfig), { join: '\n\t', skip: ['icon', 'htmlColor'] })}
+    ${
+        data.icon && data.icon !== 'undefined'
+            ? `icon={${(data.icon as string).replace('/>', '')}fontSize={'inherit'}${
+                  data.htmlColor && data.htmlColor !== 'undefined' ? ` htmlColor={'${data.htmlColor as string}'}` : ''
+              } />}`
```

Comment:
`undefined` is a fairly common thing in our react components, given that we often want to test the component behavior when an optional prop is not given. Is it possible to incorporate this into the package? Like, for all `required: false` prop data, add an extra field `undefinedOption: true` or something like that?

Reply:
I think what you are describing is achievable already. Each parameter that you define to show in the sidebar has the following core values configurable:
![image](https://github.com/etn-ccis/blui-react-component-library/assets/29152776/b58fdc14-defa-4dd0-bea3-bbd50580c21c)

So you can set the defaultValue that a prop has (for required or optional props).

The `'undefined'` string being used here is because this parameter renders as a select/dropdown and there is an explicit option in there for 'undefined' (without the option, there is no way to unselect an item once you've selected one). The select input type requires option `values` to be `number | string` — this could _probably_ be extended to support a value of `undefined` as well if we want to be able to do a comparison like `data.icon !== undefined` instead. That might complicate the typings a little bit (probably why I didn't do it initially)

Reply:
Now I think I remember — allowing `undefined` as a value in the dropdown would throw an error:
> Warning: A component is changing an uncontrolled input to be controlled. This is likely caused by the value changing from undefined to a defined value, which should not happen. Decide between using a controlled or uncontrolled input element for the lifetime of the component

Reply:
I had considered providing the ability to do a value mapping from string (label in dropdown) to anything (actual value you want to parameter to have, e.g. `undefined` or `<Component />`), like `options: [{label: 'undefined', value: undefined}]`, but that seemed to overcomplicate things for minimal gain. But we could revisit that if we want to. It might help when defining the preview component so the user doesn't have to do the mapping/lookup themselves.

Reply:
I think it can help with both 
* our testing (passing in the supposed default value vs. actually passing in undefined) and
* simplify this `generateSnippet` function that our adopters will write, like what you said there

Reply:
I took another look at it, and I think we'll leave it as-is for now. With the mapping capability, the user would still be stuck — e.g., In the case of the ChannelValue example, when picking an icon, the actual icon element they would pass as the value in the mapping would need to also know the value of the htmlColor input to render properly. Since the config object is static, they wouldn't be able to access that variable, so I don't think there's any real gain. That may not have made a lot of sense, but I can show you what I mean if you want to dig deeper.